backend/dataall/core/stacks/api/resolvers.py

```diff
@@ -64,6 +65,7 @@ def resolve_task_id(context, source: Stack, **kwargs):
         return source.EcsTaskArn.split('/')[-1]
 
 
```

Comment:
Move this check to the service layer (`StackService.get_stack_logs`). Despite being a bit polluted in this case, resolvers should just check if the input is valid and call the corresponding service.

Reply:
Check comments in `stack_logs_util` --->
we could use the decorator as:
```
def map_target_stack_to_config(target_type):
    if target_type == 'environment':
        return 'core.features.show_stack_logs'
    if target_type == 'dataset':
        return 'modules.s3_datasets.features.show_stack_logs'


@is_feature_allowed_values(resolve_config=map_target_stack_to_config, allowed_values=['enabled', 'admin_only'])
```

Reply:
@dlpzx , thanks for all the code snippets to give me a very clear understanding. I now understand how you would want to make it generic. I really like it. 

Where do you think we should keep this `map_target_stack_to_config ` function since it is mapping specifics of dataset and environment. Should I place it here ?